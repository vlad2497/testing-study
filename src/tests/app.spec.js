import React from "react";
import App from './../components/app/index';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

const setUp = () => shallow(<App/>)

describe('app component', () => {
  let component = null
  let instance = null

  beforeEach(() => {
    component = setUp()
    instance = component.instance()
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })

  describe('check input handlers', () => {
    it('onChange handler', () => {
      expect(component.state().inputValue).toBe('')
      instance.inputHandler({target: {value: 'test data'}})
      expect(component.state().inputValue).toBe('test data')
    })

    it('onKeyPress handler', () => {
      instance.setState({inputValue: 'test data'})
      instance.onKeyPressHandler({key: 'Enter'})
      expect(component.state().inputValue).toBe('')
    })
  })

  describe('check button handler', () => {
    it('check by snapshot', () => {
      const btn = component.find('.testButton')
      btn.simulate('click')
      expect(component).toMatchSnapshot()
    })

    it('check by state', () => {
      const btn = component.find('.testButton')
      btn.simulate('click')
      expect(component.state().inputValue).toBe('it is text')
    })

    it('check by instance handler', () => {
      instance.buttonHandler('it is text')
      expect(component.state().inputValue).toBe('it is text')
    })
  })
})


  // shallow - целевой

  // render - целевой + дочерние
  /*it('render snapshot', () => {
    const component = render(<App/>)
    expect(component).toMatchSnapshot()
  })*/

  // mount - целевой + дочерние + side effects
  /*it('mount snapshot', () => {
    const component = mount(<App/>)
    expect(component).toMatchSnapshot()
  })*/
