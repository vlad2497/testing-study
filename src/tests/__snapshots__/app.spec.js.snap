// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component check button handler check by snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <Input
      onChange={[Function]}
      onKeyPress={[Function]}
      value="it is text"
    />
    <button
      className="testButton"
      onClick={[Function]}
    >
      Click me!
    </button>
    <Post
      text="post"
    />
    <Info />
  </header>
</div>
`;

exports[`app component should render 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <Input
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
    <button
      className="testButton"
      onClick={[Function]}
    >
      Click me!
    </button>
    <Post
      text="post"
    />
    <Info />
  </header>
</div>
`;
